Website/src/rendering/src/components/Products/ProductSpecField.tsx

```diff
@@ -0,0 +1,54 @@
+import { RequiredDeep, Spec } from 'ordercloud-javascript-sdk';
+import { ChangeEvent, FunctionComponent } from 'react';
+
+interface ProductSpecFieldProps {
+  spec: RequiredDeep<Spec>;
+  optionId?: string;
+  value?: string;
+  onChange: (values: { SpecID: string; OptionID?: string; Value?: string }) => void;
+}
+
+const ProductSpecField: FunctionComponent<ProductSpecFieldProps> = ({
+  spec,
+  optionId,
+  value,
+  onChange,
+}) => {
+  const handleSelectChange = (e: ChangeEvent<HTMLSelectElement>) => {
+    onChange({
+      SpecID: spec.ID,
+      OptionID: e.target.value ? e.target.value : undefined,
+      Value: undefined,
+    });
+  };
+
+  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+    onChange({
+      SpecID: spec.ID,
+      OptionID: 'OpenText',
+      Value: e.target.value.length ? e.target.value : undefined,
+    });
+  };
+
+  return (
+    <div>
+      <label htmlFor={spec.ID}>
+        {spec.Name}
+        {spec.OptionCount ? (
```

Comment:
This line is another example of a condition inside the main JSX of the component.